std/process.d

```diff
@@ -806,6 +806,68 @@ private void setCLOEXEC(int fd, bool on) nothrow @nogc
     assert (wait(spawnProcess([prog.path, "foo", "bar"])) == 0);
 }
 
+version (Posix) @system unittest
+{
+    import core.sys.posix.fcntl : open, O_RDONLY;
+    import core.sys.posix.unistd : close;
+    import std.path : buildPath;
+    import std.algorithm.searching : canFind, findSplitBefore;
+    import std.functional : reverseArgs;
+    import std.array : split;
+    import std.conv : to;
+    static import std.file;
+
+    auto directory = uniqueTempPath();
+    std.file.mkdir(directory);
+    scope(exit) std.file.rmdirRecurse(directory);
+    auto path = buildPath(directory, "tmp");
+    std.file.write(path, null);
+    auto fd = open(path.tempCString, O_RDONLY);
+    scope(exit) close(fd);
+
+    // try /proc/<pid>/fd/ on linux
+    version (linux)
+    {
+        TestScript proc = "ls /proc/$$/fd";
```

Comment:
I'm concerned about this. Generally things like `$$` are shell substitutions, and won't magically replace without executing the command via a shell.

Are we sure this is actually working?